src/Microsoft.AspNet.Razor.Runtime/Runtime/TagHelpers/TagHelperExecutionContext.cs

```diff
@@ -135,7 +135,7 @@ public bool ChildContentRetrieved
         /// <summary>
         /// <see cref="ITagHelper"/>s that should be run.
         /// </summary>
-        public IEnumerable<ITagHelper> TagHelpers
+        public IList<ITagHelper> TagHelpers
```

Comment:
OIC you're updating the collection in place. Still seems incorrect for the rest of the system. Why not sort upfront and expose the sorted enumeration here?